TestDs3/Helpers/TransferItemSources/TestReadTransferItemSource.cs

```diff
@@ -224,6 +224,74 @@ public void EnumerateTransfersStreamsNewlyAvailableTransferItems()
             client.VerifyAll();
         }
 
+        [Test]
+        public void TestEnumerateTransfersSetRetryAfter()
+        {
+            var initialJobResponse = Stubs.BuildJobResponse(Stubs.Chunk1(null, false, false));
+            var factory = new Mock<IDs3ClientFactory>(MockBehavior.Strict);
+            factory
+                .Setup(f => f.GetClientForNodeId(It.IsAny<Guid?>()))
+                .Returns(new Mock<IDs3Client>(MockBehavior.Strict).Object);
+            var client = new Mock<IDs3Client>(MockBehavior.Strict);
+            client
+                .Setup(c => c.BuildFactory(It.IsAny<IEnumerable<Node>>()))
+                .Returns(factory.Object);
+            client
+                .Setup(c => c.GetAvailableJobChunks(AvailableChunks(Stubs.JobId)))
+                .Returns(GetAvailableJobChunksResponse.RetryAfter(
+                    TimeSpan.FromMinutes(5)));
+
+            var transferItemSource = new ReadTransferItemSource(_ => { }, client.Object, 2, initialJobResponse);
+
+            using (var transfers = transferItemSource.EnumerateAvailableTransfers().GetEnumerator())
+            {
+                try
+                {
+                    Assert.True(transferItemSource.RetryAfterLeft == 2);
+                    transfers.MoveNext();
+                }
+                catch (Ds3NoMoreRetriesException ex)
+                {
+                    Assert.True(transferItemSource.RetryAfterLeft == 0);
+                    Assert.True(ex.Message.Equals("Reached the limit number of retries request"));
+                    return;
+                }
+                Assert.Fail();
```

Comment:
Why not move this to right after line 251.  That way you don't have to have the return and it's very clear what the failure condition is.